app/lib/service/entrypoint/_isolate.dart

```diff
@@ -139,23 +123,67 @@ class IsolateGroup {
     }
   }
 
-  /// Starts [count] new isolates, and after a [wait] duration,
-  /// closes the old ones.
+  /// Starts [count] new isolates, waits for the pending requests to get processed,
+  /// and after a maximum [wait] duration, closes the old ones.
   Future<void> renew({
     required int count,
```

Comment:
Is it not extremely dangerous that we can pass a different `count` to `renew` and `start`.

What is the behavior if this happens? is this fully thought through? or are we assuming that `count` will always have same value when a specific instance of `IsolateGroup` is called?